frontend/src/component/layout/MainLayout/NavigationSidebar/NavigationSidebar.tsx

```diff
@@ -247,29 +298,137 @@ const useNavigationMode = () => {
     return [mode, setMode] as const;
 };
 
-export const NavigationSidebar = () => {
-    const { routes, showBadge } = useRoutes();
-
-    const [mode, setMode] = useNavigationMode();
-
-    const DynamicListItem = mode === 'mini' ? MiniListItem : FullListItem;
+const MainNavigationList: FC<{ mode: 'mini' | 'full'; onClick?: () => void }> =
+    ({ mode, onClick }) => {
+        const DynamicListItem = mode === 'mini' ? MiniListItem : FullListItem;
 
-    return (
-        <StyledBox>
+        return (
             <List>
-                <DynamicListItem href='/projects' text='Projects'>
+                <DynamicListItem
+                    href='/projects'
+                    text='Projects'
+                    onClick={onClick}
+                >
                     <StyledProjectIcon />
                 </DynamicListItem>
-                <DynamicListItem href='/search' text='Search'>
+                <DynamicListItem href='/search' text='Search' onClick={onClick}>
                     <SearchIcon />
                 </DynamicListItem>
-                <DynamicListItem href='/playground' text='Playground'>
+                <DynamicListItem
+                    href='/playground'
+                    text='Playground'
+                    onClick={onClick}
+                >
                     <PlaygroundIcon />
                 </DynamicListItem>
-                <DynamicListItem href='/insights' text='Insights'>
+                <DynamicListItem
+                    href='/insights'
+                    text='Insights'
+                    onClick={onClick}
+                >
                     <InsightsIcon />
                 </DynamicListItem>
             </List>
+        );
+    };
+
+const ConfigureNavigationList: FC<{
```

Comment:
Extracting to new component so that we can use it in desktop version and mobile version